main/src/ca/uwaterloo/flix/language/phase/EffectBinder.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2023 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.Ast.BoundBy
+import ca.uwaterloo.flix.language.ast.ReducedAst._
+import ca.uwaterloo.flix.language.ast._
+import ca.uwaterloo.flix.util.collection.Chain
+import ca.uwaterloo.flix.util.{InternalCompilerException, ParOps}
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+
+object EffectBinder {
+
+  /**
+    * Identity Function.
+    */
+  def run(root: Root)(implicit flix: Flix): Root = flix.phase("EffectBinder") {
```

Comment:
Can we come up with a better name than EffectBinder? I guess its not (yet) FG CBV?

Reply:
StacklessEffects?